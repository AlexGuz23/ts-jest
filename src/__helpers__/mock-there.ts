export default function mockThese(map: string[] | { [k: string]: () => any }) {
  const isArray = Array.isArray(map);
  const items: string[] = isArray ? (map as string[]) : Object.keys(map);
  items.forEach(item => {
    const val = isArray ? () => item : map[item];
    jest.doMock(item, val, { virtual: true });
  });
}
