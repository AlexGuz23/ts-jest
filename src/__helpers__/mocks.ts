import { rootLogger } from '../util/logger'
import { testing } from 'bs-logger'

// typings helper
export function mocked<T>(
  val: T,
): T extends (...args: any[]) => any ? jest.MockInstance<T> : jest.Mocked<T> {
  return val as any
}
export function spied<T>(
  val: T,
): T extends (...args: any[]) => any ? jest.SpyInstance<T> : jest.Mocked<T> {
  return val as any
}

export const logTargetMock = () => (rootLogger as testing.LoggerMock).target
