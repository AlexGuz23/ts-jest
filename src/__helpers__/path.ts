import { resolve, relative } from 'path'

export const ROOT = resolve(__dirname, '..', '..')

export function relativeToRoot(path: string) {
  return relative(ROOT, path)
}
