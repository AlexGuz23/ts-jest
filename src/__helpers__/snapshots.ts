import outdent from 'outdent'

export const snap = outdent({
  trimTrailingNewline: false,
  trimLeadingNewline: false,
})
