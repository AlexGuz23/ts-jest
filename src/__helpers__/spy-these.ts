export default function spyThese<T extends object, K extends keyof T>(
  object: T,
  implementations: { [key in K]: T[K] | any | undefined },
): { [key in K]: jest.SpyInstance<T[K]> } & { mockRestore: () => void } {
  const keys = Object.keys(implementations) as K[];
  const res = keys.reduce(
    (map, key) => {
      const actual = object[key] as any;
      const spy = jest.spyOn(object, key as K);
      if (implementations[key]) {
        const impl = implementations[key] as (...args: any[]) => any;
        if (impl.length && /\W\$super\W/.test(impl.toString())) {
          spy.mockImplementation(function(this: T, ...args: any[]) {
            return impl.call(this, () => actual.apply(this, args), ...args);
          });
        } else {
          spy.mockImplementation(impl);
        }
      }
      return map;
    },
    {} as any,
  );
  // utility to restore all
  res.mockRestore = () => {
    keys.forEach(key => res[key].mockRestore());
  };
  return res;
}
