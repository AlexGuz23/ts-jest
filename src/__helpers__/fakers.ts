import { TsJestGlobalOptions, BabelConfig } from '../types';
import { resolve } from 'path';
import { ImportReasons } from '../utils/messages';

export function filePath(relPath: string): string {
  return resolve(__dirname, '..', '..', relPath);
}

export const rootDir = filePath('');

export function transpiledTsSource() {
  return `
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var upper_1 = __importDefault(require("./upper"));
var lower_1 = __importDefault(require("./lower"));
jest.mock('./upper', function () { return function (s) { return s.toUpperCase(); }; });
describe('hello', function () {
    test('my test', function () {
        expect(upper_1.default('hello')).toBe('HELLO');
        expect(lower_1.default('HELLO')).toBe('hello');
        jest.mock('./lower', function () { return function (s) { return s.toLowerCase(); }; });
    });
});
`;
}

export function htmlSource() {
  return `
<div>
  <span>some text with \`backtick\`</span>
</div>
`;
}

export function typescriptSource() {
  return `
import upper from './upper';
import lower from './lower';

jest.mock('./upper', () => (s) => s.toUpperCase());

describe('hello', () => {
  test('my test', () => {
    expect(upper('hello')).toBe('HELLO');
    expect(lower('HELLO')).toBe('hello');
    jest.mock('./lower', () => (s) => s.toLowerCase());
  });
});
`;
}

export function tsJestConfig<T extends TsJestGlobalOptions>(
  options?: TsJestGlobalOptions,
): T {
  return { ...options } as any;
}

export function jestConfig<T extends jest.ProjectConfig>(
  options?: jest.InitialOptions,
  tsJestOptions?: TsJestGlobalOptions,
): T {
  const res = {
    globals: {},
    moduleFileExtensions: ['ts', 'js'],
    ...options,
  } as any;
  if (tsJestOptions) {
    res.globals['ts-jest'] = tsJestConfig(tsJestOptions);
  }
  return res;
}

export function babelConfig<T extends BabelConfig>(options?: BabelConfig): T {
  return {
    ...options,
    presets: [...(options && options.presets)],
    plugins: [...(options && options.plugins)],
  } as any;
}

export function importReason(text: string = 'because'): ImportReasons {
  return text as any;
}
