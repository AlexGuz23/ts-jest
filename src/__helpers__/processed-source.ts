import { RawSourceMap } from 'source-map'
import {
  rewriteSourceMaps,
  relativisePaths,
  parseSource,
  ParsedSourceWithMaps,
} from './source-maps'
import { ROOT } from './path'
import { isAbsolute, relative } from 'path'

// tslint:disable-next-line:no-default-export
export default class ProcessedSource {
  readonly filename: string

  constructor(
    readonly output: string,
    filename: string,
    readonly cwd: string = ROOT,
  ) {
    if (isAbsolute(filename)) filename = relative(this.cwd, filename)
    this.filename = filename
  }
  get outputCode(): string {
    return this.output
  }
  get parsedSource(): ParsedSourceWithMaps {
    return parseSource(this.output)
  }
  get outputCodeWithoutMaps(): string {
    return this.parsedSource.source
  }
  get outputSourceMaps(): RawSourceMap | undefined {
    return this.parsedSource.sourceMaps
  }
  get normalizedOutputCode(): string {
    return rewriteSourceMaps(this.output, this.sourceMapsNormalizer)
  }
  get normalizedOutputSourceMaps(): RawSourceMap | undefined {
    const maps = this.outputSourceMaps
    if (maps) return this.sourceMapsNormalizer(maps)
    return
  }
  get sourceMapsNormalizer() {
    return (maps: RawSourceMap): RawSourceMap =>
      relativisePaths(maps, this.cwd, '<cwd>/')
  }
}
