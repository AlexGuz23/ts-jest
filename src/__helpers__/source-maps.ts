// WARNING: this file is shared between e2e and unit tests
import { RawSourceMap } from 'source-map'
import bufferFrom from 'buffer-from'
import { relative, isAbsolute, resolve } from 'path'
import stableStringify = require('fast-json-stable-stringify')
import { realpathSync } from 'fs'

// source-map module doesn't provide a sync method to extract the source-maps
export function extractSourceMaps(source: string): RawSourceMap | undefined {
  const [, comment]: [any, string | undefined] =
    (source.match(
      /[\n^]\/\/#\s*sourceMappingURL=data:application\/json;(?:charset=utf-8;)?base64,(\S+)\s*$/,
    ) as any) || []
  if (!comment) return

  return base64ToSourceMaps(comment)
}

export function base64ToSourceMaps(base64: string): RawSourceMap {
  return JSON.parse(bufferFrom(base64, 'base64').toString('utf8'))
}

export function sourceMapsToBase64(sourceMaps: RawSourceMap): string {
  return bufferFrom(stableStringify(sourceMaps)).toString('base64')
}

export function relativiseSourceRoot(
  fromPath: string,
  source: string,
  prefix: string = '',
): string {
  const from = realpathSync(fromPath)
  const remap = (path: string): string =>
    (isAbsolute(path)
      ? `${prefix}${relative(from, realpathSync(path))}`
      : path
    ).replace(/\\/g, '/')

  return source.replace(
    /([\n^]\/\/#\s*sourceMappingURL=data:application\/json;(?:charset=utf-8;)?base64,)(\S+)(\s*)$/,
    (_, before, base64, after) => {
      const map = base64ToSourceMaps(base64)
      if (map.sourceRoot) map.sourceRoot = remap(map.sourceRoot)
      if (map.sources) map.sources = map.sources.map(remap)
      if (map.file) map.file = remap(map.file)
      return `${before}${sourceMapsToBase64(map)}${after}`
    },
  )
}
