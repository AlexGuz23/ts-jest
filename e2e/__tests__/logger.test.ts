import { configureTestCase } from '../__helpers__/test-case'
import { PackageSets, allValidPackageSets } from '../__helpers__/templates'
import { existsSync } from 'fs'

describe('With unsupported version test', () => {
  const testCase = configureTestCase('simple')

  testCase.runWithTemplates([PackageSets.unsupportedVersion], 0, (runTest, { testLabel }) => {
    it(testLabel, () => {
      const result = runTest()
      expect(result.status).toBe(0)
      expect(result).toMatchSnapshot()
    })
  })
})

describe('TS_JEST_DEBUG', () => {
  const testCase = configureTestCase('simple', { env: { TS_JEST_DEBUG: 'true' } })

  testCase.runWithTemplates(allValidPackageSets, 0, (runTest, { templateName }) => {
    it(`should pass and create log file when using tempalte "${templateName}"`, () => {
      const result = runTest()
      expect(result.status).toBe(0)
      expect(existsSync(result.logFilePath))
      expect(result.normalizedLogFileContent).toMatchSnapshot()
    })
  })
})
