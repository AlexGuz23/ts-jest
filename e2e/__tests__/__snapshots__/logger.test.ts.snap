// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TS_JEST_DEBUG should pass and create log file when using tempalte "default" 1`] = `
"initiailized
Importer.instance creating Importer singleton
checkVersion jest OK { actual: '23.4.2', expected: '>=22 <24' }
created new transformer [object TsJestTransformer<#1>]
backportJestConfig
Importer#tryThese loaded module typescript
Importer#_patch patching typescript
checkVersion typescript OK { actual: '3.0.1', expected: '>=2.7 <4' }
createCompiler without type-checking, config: { version: '24.0.0-beta.0',\\\\n  jest:\\\\n   { automock: false,\\\\n     browser: false,\\\\n     cache: true,\\\\n     clearMocks: false,\\\\n     coveragePathIgnorePatterns: [ '/node_modules/' ],\\\\n     cwd:\\\\n      '<cwd>',\\\\n     detectLeaks: false,\\\\n     detectOpenHandles: false,\\\\n     errorOnDeprecated: false,\\\\n     filter: null,\\\\n     forceCoverageMatch: [],\\\\n     globals: {},\\\\n     haste: { providesModuleNodeModules: [] },\\\\n     moduleDirectories: [ 'node_modules' ],\\\\n     moduleFileExtensions: [ 'js', 'json', 'jsx', 'node', 'ts', 'tsx' ],\\\\n     moduleNameMapper: {},\\\\n     modulePathIgnorePatterns: [],\\\\n     prettierPath: null,\\\\n     resetMocks: false,\\\\n     resetModules: false,\\\\n     resolver: null,\\\\n     restoreMocks: false,\\\\n     rootDir:\\\\n      '<cwd>',\\\\n     roots:\\\\n      [ '<cwd>' ],\\\\n     runner: 'jest-runner',\\\\n     setupFiles: [],\\\\n     setupTestFrameworkScriptFile: null,\\\\n     skipFilter: false,\\\\n     snapshotSerializers: [],\\\\n     testEnvironment:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/default/node_modules/jest-environment-node/build/index.js',\\\\n     testEnvironmentOptions: {},\\\\n     testLocationInResults: false,\\\\n     testMatch:\\\\n      [ '**/__tests__/**/*.js?(x)',\\\\n        '**/?(*.)+(spec|test).js?(x)',\\\\n        '**/__tests__/**/*.ts?(x)',\\\\n        '**/?(*.)+(spec|test).ts?(x)' ],\\\\n     testPathIgnorePatterns: [ '/node_modules/' ],\\\\n     testRegex: '',\\\\n     testRunner:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/default/node_modules/jest-jasmine2/build/index.js',\\\\n     testURL: 'about:blank',\\\\n     timers: 'real',\\\\n     transform: [ [Array] ],\\\\n     transformIgnorePatterns: [ '/node_modules/' ],\\\\n     watchPathIgnorePatterns: [] },\\\\n  tsJest:\\\\n   { tsConfig: { kind: 'inline', value: {} },\\\\n     babelConfig: undefined,\\\\n     diagnostics: { ignoreCodes: [Array], pretty: true },\\\\n     typeCheck: false,\\\\n     compiler: 'typescript',\\\\n     stringifyContentPathRegex: undefined },\\\\n  babel: undefined,\\\\n  tsconfig: undefined }
readThrough:cache-miss <cwd>/Hello.spec.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.spec.ts
readThrough:write-caches <cwd>/Hello.spec.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
readThrough:cache-miss <cwd>/Hello.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.ts
readThrough:write-caches <cwd>/Hello.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
"
`;

exports[`TS_JEST_DEBUG should pass and create log file when using tempalte "with-babel-6" 1`] = `
"initiailized
Importer.instance creating Importer singleton
checkVersion jest OK { actual: '23.4.2', expected: '>=22 <24' }
created new transformer [object TsJestTransformer<#1>]
backportJestConfig
Importer#tryThese loaded module babel-core
Importer#_patch patching babel-core
checkVersion babel-core OK { actual: '6.26.3', expected: '>=6 <7 || 7.0.0-bridge.0' }
Importer#tryThese loaded module typescript
Importer#_patch patching typescript
checkVersion typescript OK { actual: '3.0.1', expected: '>=2.7 <4' }
Importer#tryThese loaded module babel-core
Importer#tryThese loaded module babel-jest
Importer#_patch patching babel-jest
checkVersion babel-jest OK { actual: '23.4.2', expected: '>=22 <24' }
createCompiler without type-checking, config: { version: '24.0.0-beta.0',\\\\n  jest:\\\\n   { automock: false,\\\\n     browser: false,\\\\n     cache: true,\\\\n     clearMocks: false,\\\\n     coveragePathIgnorePatterns: [ '/node_modules/' ],\\\\n     cwd:\\\\n      '<cwd>',\\\\n     detectLeaks: false,\\\\n     detectOpenHandles: false,\\\\n     errorOnDeprecated: false,\\\\n     filter: null,\\\\n     forceCoverageMatch: [],\\\\n     globals: {},\\\\n     haste: { providesModuleNodeModules: [] },\\\\n     moduleDirectories: [ 'node_modules' ],\\\\n     moduleFileExtensions: [ 'js', 'json', 'jsx', 'node', 'ts', 'tsx' ],\\\\n     moduleNameMapper: {},\\\\n     modulePathIgnorePatterns: [],\\\\n     prettierPath: null,\\\\n     resetMocks: false,\\\\n     resetModules: false,\\\\n     resolver: null,\\\\n     restoreMocks: false,\\\\n     rootDir:\\\\n      '<cwd>',\\\\n     roots:\\\\n      [ '<cwd>' ],\\\\n     runner: 'jest-runner',\\\\n     setupFiles: [],\\\\n     setupTestFrameworkScriptFile: null,\\\\n     skipFilter: false,\\\\n     snapshotSerializers: [],\\\\n     testEnvironment:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/with-babel-6/node_modules/jest-environment-node/build/index.js',\\\\n     testEnvironmentOptions: {},\\\\n     testLocationInResults: false,\\\\n     testMatch:\\\\n      [ '**/__tests__/**/*.js?(x)',\\\\n        '**/?(*.)+(spec|test).js?(x)',\\\\n        '**/__tests__/**/*.ts?(x)',\\\\n        '**/?(*.)+(spec|test).ts?(x)' ],\\\\n     testPathIgnorePatterns: [ '/node_modules/' ],\\\\n     testRegex: '',\\\\n     testRunner:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/with-babel-6/node_modules/jest-jasmine2/build/index.js',\\\\n     testURL: 'about:blank',\\\\n     timers: 'real',\\\\n     transform: [ [Array] ],\\\\n     transformIgnorePatterns: [ '/node_modules/' ],\\\\n     watchPathIgnorePatterns: [] },\\\\n  tsJest:\\\\n   { tsConfig: { kind: 'inline', value: {} },\\\\n     babelConfig: { kind: 'file', value: undefined },\\\\n     diagnostics: { ignoreCodes: [Array], pretty: true },\\\\n     typeCheck: false,\\\\n     compiler: 'typescript',\\\\n     stringifyContentPathRegex: undefined },\\\\n  babel:\\\\n   { filename: 'unknown',\\\\n     filenameRelative: undefined,\\\\n     inputSourceMap: undefined,\\\\n     env: {},\\\\n     mode: undefined,\\\\n     retainLines: false,\\\\n     highlightCode: true,\\\\n     suppressDeprecationMessages: false,\\\\n     presets: [],\\\\n     plugins: [],\\\\n     ignore: [],\\\\n     only: undefined,\\\\n     code: true,\\\\n     metadata: true,\\\\n     ast: true,\\\\n     extends: undefined,\\\\n     comments: true,\\\\n     shouldPrintComment: undefined,\\\\n     wrapPluginVisitorMethod: undefined,\\\\n     compact: 'auto',\\\\n     minified: false,\\\\n     sourceMap: undefined,\\\\n     sourceMaps: undefined,\\\\n     sourceMapTarget: undefined,\\\\n     sourceFileName: undefined,\\\\n     sourceRoot: undefined,\\\\n     babelrc: true,\\\\n     sourceType: 'module',\\\\n     auxiliaryCommentBefore: undefined,\\\\n     auxiliaryCommentAfter: undefined,\\\\n     resolveModuleSource: undefined,\\\\n     getModuleId: undefined,\\\\n     moduleRoot: undefined,\\\\n     moduleIds: false,\\\\n     moduleId: undefined,\\\\n     passPerPreset: false,\\\\n     parserOpts: false,\\\\n     generatorOpts: false },\\\\n  tsconfig: undefined }
readThrough:cache-miss <cwd>/Hello.spec.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.spec.ts
readThrough:write-caches <cwd>/Hello.spec.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
readThrough:cache-miss <cwd>/Hello.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.ts
readThrough:write-caches <cwd>/Hello.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
"
`;

exports[`TS_JEST_DEBUG should pass and create log file when using tempalte "with-babel-7" 1`] = `
"initiailized
Importer.instance creating Importer singleton
checkVersion jest OK { actual: '23.4.2', expected: '>=22 <24' }
created new transformer [object TsJestTransformer<#1>]
backportJestConfig
Importer#tryThese loaded module babel-core
Importer#_patch patching babel-core
checkVersion babel-core OK { actual: '7.0.0-bridge.0',\\\\n  expected: '>=6 <7 || 7.0.0-bridge.0' }
Importer#tryThese loaded module typescript
Importer#_patch patching typescript
checkVersion typescript OK { actual: '3.0.1', expected: '>=2.7 <4' }
Importer#tryThese loaded module babel-core
Importer#tryThese loaded module babel-jest
Importer#_patch patching babel-jest
checkVersion babel-jest OK { actual: '23.4.2', expected: '>=22 <24' }
createCompiler without type-checking, config: { version: '24.0.0-beta.0',\\\\n  jest:\\\\n   { automock: false,\\\\n     browser: false,\\\\n     cache: true,\\\\n     clearMocks: false,\\\\n     coveragePathIgnorePatterns: [ '/node_modules/' ],\\\\n     cwd:\\\\n      '<cwd>',\\\\n     detectLeaks: false,\\\\n     detectOpenHandles: false,\\\\n     errorOnDeprecated: false,\\\\n     filter: null,\\\\n     forceCoverageMatch: [],\\\\n     globals: {},\\\\n     haste: { providesModuleNodeModules: [] },\\\\n     moduleDirectories: [ 'node_modules' ],\\\\n     moduleFileExtensions: [ 'js', 'json', 'jsx', 'node', 'ts', 'tsx' ],\\\\n     moduleNameMapper: {},\\\\n     modulePathIgnorePatterns: [],\\\\n     prettierPath: null,\\\\n     resetMocks: false,\\\\n     resetModules: false,\\\\n     resolver: null,\\\\n     restoreMocks: false,\\\\n     rootDir:\\\\n      '<cwd>',\\\\n     roots:\\\\n      [ '<cwd>' ],\\\\n     runner: 'jest-runner',\\\\n     setupFiles: [],\\\\n     setupTestFrameworkScriptFile: null,\\\\n     skipFilter: false,\\\\n     snapshotSerializers: [],\\\\n     testEnvironment:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/with-babel-7/node_modules/jest-environment-node/build/index.js',\\\\n     testEnvironmentOptions: {},\\\\n     testLocationInResults: false,\\\\n     testMatch:\\\\n      [ '**/__tests__/**/*.js?(x)',\\\\n        '**/?(*.)+(spec|test).js?(x)',\\\\n        '**/__tests__/**/*.ts?(x)',\\\\n        '**/?(*.)+(spec|test).ts?(x)' ],\\\\n     testPathIgnorePatterns: [ '/node_modules/' ],\\\\n     testRegex: '',\\\\n     testRunner:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/with-babel-7/node_modules/jest-jasmine2/build/index.js',\\\\n     testURL: 'about:blank',\\\\n     timers: 'real',\\\\n     transform: [ [Array] ],\\\\n     transformIgnorePatterns: [ '/node_modules/' ],\\\\n     watchPathIgnorePatterns: [] },\\\\n  tsJest:\\\\n   { tsConfig: { kind: 'inline', value: {} },\\\\n     babelConfig: { kind: 'file', value: undefined },\\\\n     diagnostics: { ignoreCodes: [Array], pretty: true },\\\\n     typeCheck: false,\\\\n     compiler: 'typescript',\\\\n     stringifyContentPathRegex: undefined },\\\\n  babel:\\\\n   { cwd:\\\\n      '<cwd>',\\\\n     babelrc: false,\\\\n     configFile: false,\\\\n     passPerPreset: false,\\\\n     envName: 'test',\\\\n     root:\\\\n      '<cwd>',\\\\n     plugins: [],\\\\n     presets: [] },\\\\n  tsconfig: undefined }
readThrough:cache-miss <cwd>/Hello.spec.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.spec.ts
readThrough:write-caches <cwd>/Hello.spec.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
readThrough:cache-miss <cwd>/Hello.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.ts
readThrough:write-caches <cwd>/Hello.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
"
`;

exports[`TS_JEST_DEBUG should pass and create log file when using tempalte "with-jest-22" 1`] = `
"initiailized
Importer.instance creating Importer singleton
checkVersion jest OK { actual: '22.4.4', expected: '>=22 <24' }
created new transformer [object TsJestTransformer<#1>]
backportJestConfig
Importer#tryThese loaded module typescript
Importer#_patch patching typescript
checkVersion typescript OK { actual: '3.0.1', expected: '>=2.7 <4' }
createCompiler without type-checking, config: { version: '24.0.0-beta.0',\\\\n  jest:\\\\n   { automock: false,\\\\n     browser: false,\\\\n     cache: true,\\\\n     clearMocks: false,\\\\n     coveragePathIgnorePatterns: [ '/node_modules/' ],\\\\n     cwd:\\\\n      '<cwd>',\\\\n     detectLeaks: false,\\\\n     forceCoverageMatch: [],\\\\n     globals: {},\\\\n     haste: { providesModuleNodeModules: [] },\\\\n     moduleDirectories: [ 'node_modules' ],\\\\n     moduleFileExtensions: [ 'js', 'json', 'jsx', 'node', 'ts', 'tsx' ],\\\\n     moduleNameMapper: {},\\\\n     modulePathIgnorePatterns: [],\\\\n     resetMocks: false,\\\\n     resetModules: false,\\\\n     restoreMocks: false,\\\\n     rootDir:\\\\n      '<cwd>',\\\\n     roots:\\\\n      [ '<cwd>' ],\\\\n     runner: 'jest-runner',\\\\n     setupFiles: [],\\\\n     snapshotSerializers: [],\\\\n     testEnvironment:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/with-jest-22/node_modules/jest-environment-node/build/index.js',\\\\n     testEnvironmentOptions: {},\\\\n     testLocationInResults: false,\\\\n     testMatch:\\\\n      [ '**/__tests__/**/*.js?(x)',\\\\n        '**/?(*.)+(spec|test).js?(x)',\\\\n        '**/__tests__/**/*.ts?(x)',\\\\n        '**/?(*.)+(spec|test).ts?(x)' ],\\\\n     testPathIgnorePatterns: [ '/node_modules/' ],\\\\n     testRegex: '',\\\\n     testRunner:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/with-jest-22/node_modules/jest-jasmine2/build/index.js',\\\\n     testURL: 'about:blank',\\\\n     timers: 'real',\\\\n     transform: [ [Array] ],\\\\n     transformIgnorePatterns: [ '/node_modules/' ],\\\\n     watchPathIgnorePatterns: [] },\\\\n  tsJest:\\\\n   { tsConfig: { kind: 'inline', value: {} },\\\\n     babelConfig: undefined,\\\\n     diagnostics: { ignoreCodes: [Array], pretty: true },\\\\n     typeCheck: false,\\\\n     compiler: 'typescript',\\\\n     stringifyContentPathRegex: undefined },\\\\n  babel: undefined,\\\\n  tsconfig: undefined }
readThrough:cache-miss <cwd>/Hello.spec.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.spec.ts
readThrough:write-caches <cwd>/Hello.spec.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
readThrough:cache-miss <cwd>/Hello.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.ts
readThrough:write-caches <cwd>/Hello.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
"
`;

exports[`TS_JEST_DEBUG should pass and create log file when using tempalte "with-typescript-2-7" 1`] = `
"initiailized
Importer.instance creating Importer singleton
checkVersion jest OK { actual: '23.4.2', expected: '>=22 <24' }
created new transformer [object TsJestTransformer<#1>]
backportJestConfig
Importer#tryThese loaded module typescript
Importer#_patch patching typescript
checkVersion typescript OK { actual: '2.7.2', expected: '>=2.7 <4' }
createCompiler without type-checking, config: { version: '24.0.0-beta.0',\\\\n  jest:\\\\n   { automock: false,\\\\n     browser: false,\\\\n     cache: true,\\\\n     clearMocks: false,\\\\n     coveragePathIgnorePatterns: [ '/node_modules/' ],\\\\n     cwd:\\\\n      '<cwd>',\\\\n     detectLeaks: false,\\\\n     detectOpenHandles: false,\\\\n     errorOnDeprecated: false,\\\\n     filter: null,\\\\n     forceCoverageMatch: [],\\\\n     globals: {},\\\\n     haste: { providesModuleNodeModules: [] },\\\\n     moduleDirectories: [ 'node_modules' ],\\\\n     moduleFileExtensions: [ 'js', 'json', 'jsx', 'node', 'ts', 'tsx' ],\\\\n     moduleNameMapper: {},\\\\n     modulePathIgnorePatterns: [],\\\\n     prettierPath: null,\\\\n     resetMocks: false,\\\\n     resetModules: false,\\\\n     resolver: null,\\\\n     restoreMocks: false,\\\\n     rootDir:\\\\n      '<cwd>',\\\\n     roots:\\\\n      [ '<cwd>' ],\\\\n     runner: 'jest-runner',\\\\n     setupFiles: [],\\\\n     setupTestFrameworkScriptFile: null,\\\\n     skipFilter: false,\\\\n     snapshotSerializers: [],\\\\n     testEnvironment:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/with-typescript-2-7/node_modules/jest-environment-node/build/index.js',\\\\n     testEnvironmentOptions: {},\\\\n     testLocationInResults: false,\\\\n     testMatch:\\\\n      [ '**/__tests__/**/*.js?(x)',\\\\n        '**/?(*.)+(spec|test).js?(x)',\\\\n        '**/__tests__/**/*.ts?(x)',\\\\n        '**/?(*.)+(spec|test).ts?(x)' ],\\\\n     testPathIgnorePatterns: [ '/node_modules/' ],\\\\n     testRegex: '',\\\\n     testRunner:\\\\n      '<tmp>/--ts-jest-temp-e2e--/__templates__/with-typescript-2-7/node_modules/jest-jasmine2/build/index.js',\\\\n     testURL: 'about:blank',\\\\n     timers: 'real',\\\\n     transform: [ [Array] ],\\\\n     transformIgnorePatterns: [ '/node_modules/' ],\\\\n     watchPathIgnorePatterns: [] },\\\\n  tsJest:\\\\n   { tsConfig: { kind: 'inline', value: {} },\\\\n     babelConfig: undefined,\\\\n     diagnostics: { pretty: true, ignoreCodes: [Array], pathRegex: undefined },\\\\n     typeCheck: false,\\\\n     compiler: 'typescript',\\\\n     stringifyContentPathRegex: undefined },\\\\n  babel: undefined,\\\\n  tsconfig: undefined }
readThrough:cache-miss <cwd>/Hello.spec.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.spec.ts
readThrough:write-caches <cwd>/Hello.spec.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
readThrough:cache-miss <cwd>/Hello.ts
compiler#getOutput compiling as isolated module
customTranformer#hoisting <cwd>/Hello.ts
readThrough:write-caches <cwd>/Hello.ts cache file <tmp>/jest_dx/ts-jest-<hex:40>/<hex:40>.js
"
`;

exports[`With unsupported version test should pass using template "with-unsupported-version" 1`] = `
  √ jest --runInBand
  ↳ exit code: 0
  ===[ STDOUT ]===================================================================
  
  ===[ STDERR ]===================================================================
  ts-jest: Version 2.5.3 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=2.7.0 <4.0.0). Please do not report issues in ts-jest if you are using unsupported versions.
  PASS ./Hello.spec.ts
    Hello Class
      √ should create a new Hello
  
  Test Suites: 1 passed, 1 total
  Tests:       1 passed, 1 total
  Snapshots:   0 total
  Time:        XXs
  Ran all test suites.
  ================================================================================
`;
